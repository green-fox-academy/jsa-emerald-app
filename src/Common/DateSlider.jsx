import React from 'react';
import { View, Text } from 'react-native';
import { Button } from 'react-native-elements';
import PropTypes from 'prop-types';
import styles from './themeStyle';

export default function DateSlider(props) {
  const { viewSet, onPressBtn, viewType } = props;
  const format = viewType === 'month' ? 'MMM YYYY' : 'YYYY';

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
      <Button
        containerStyle={styles.dateNormal}
        title={viewSet[0].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[0], viewType)}
      />
      <View style={styles.dateHighlight}>
        <Text style={{ textAlign: 'center', fontSize: 18, fontWeight: '500' }}>{viewSet[1].format(format)}</Text>
      </View>
      <Button
        containerStyle={styles.dateNormal}
        title={viewSet[2].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[2], viewType)}
      />
    </View>
  );
}

DateSlider.propTypes = {
  viewSet: PropTypes.instanceOf(Array),
  onPressBtn: PropTypes.func.isRequired,
  viewType: PropTypes.oneOf(['month', 'year']),
};

DateSlider.defaultProps = {
  viewSet: [],
  viewType: 'month',
};
