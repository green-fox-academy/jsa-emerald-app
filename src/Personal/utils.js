function emailValidation(email) {
  // eslint-disable-next-line no-useless-escape
  const validateEmail = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return validateEmail.test(String(email).toLowerCase());
}

function passwordValidation(password) {
  const validatePassword = /^[0-9a-zA-Z]{8,16}$/;
  return validatePassword.test(password);
}

export default {
  emailValidation,
  passwordValidation,
};
