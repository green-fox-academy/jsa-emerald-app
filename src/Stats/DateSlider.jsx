import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import { Button } from 'react-native-elements';

export default function DateSlider(props) {
  const { viewSet, onPressBtn, viewType } = props;
  const format = viewType === 'month' ? 'MMM YYYY' : 'YYYY';

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-around' }}>
      <Button
        containerStyle={{ flex: 1, borderBottomWidth: 2, borderBottomColor: 'white' }}
        title={viewSet[0].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[0], viewType)}
      />
      <View style={{
        flex: 1, borderBottomWidth: 2, borderBottomColor: '#30d29d', justifyContent: 'center',
      }}
      >
        <Text style={{ textAlign: 'center', fontSize: 18, fontWeight: '500' }}>{viewSet[1].format(format)}</Text>
      </View>
      <Button
        containerStyle={{ flex: 1, borderBottomWidth: 2, borderBottomColor: 'white' }}
        title={viewSet[2].format(format)}
        type="clear"
        titleStyle={{ color: 'grey' }}
        onPress={() => onPressBtn(viewSet[2], viewType)}
      />
    </View>
  );
}

DateSlider.propTypes = {
  viewSet: PropTypes.instanceOf(Array),
  onPressBtn: PropTypes.func.isRequired,
  viewType: PropTypes.oneOf(['month', 'year']),
};

DateSlider.defaultProps = {
  viewSet: [],
  viewType: 'month',
};
