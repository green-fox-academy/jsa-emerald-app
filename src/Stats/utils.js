const sortDataByDate = (dataList) => {
  const originalDataList = dataList.slice(0);
  const orderedDataListByDateFromNearToFar = originalDataList.sort((a, b) => (b.date - a.date));
  return orderedDataListByDateFromNearToFar;
};

export default {
  sortDataByDate,
};
