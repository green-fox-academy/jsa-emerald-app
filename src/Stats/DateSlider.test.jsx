import React from 'react';
import renderer from 'react-test-renderer';
import DateSlider from './DateSlider';
import utils from './utils';

const moment = require('moment');

describe('<DateSlider />', () => {
  it('render', () => {
    const tree = renderer.create(<DateSlider
      viewSet={utils.getDateSet(moment(), 'month')}
      onPressBtn={(set, cate) => utils.getDateSet(set, cate)}
      viewType="month"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<DateSlider />', () => {
  it('render', () => {
    const type = 'year';
    const dateSet = utils.getDateSet(moment(), type);
    const tree = renderer.create(<DateSlider
      viewSet={dateSet}
      onPressBtn={(set, cate) => utils.getDateSet(set, cate)}
      viewType="year"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
