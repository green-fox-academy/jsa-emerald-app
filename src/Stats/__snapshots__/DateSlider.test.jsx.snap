// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateSlider /> render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 2,
        "borderRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Nov 2019
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#30d29d",
        "borderBottomWidth": 2,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "500",
          "textAlign": "center",
        }
      }
    >
      Dec 2019
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 2,
        "borderRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          Jan 2020
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<DateSlider /> render 2`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 2,
        "borderRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          2018
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#30d29d",
        "borderBottomWidth": 2,
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "500",
          "textAlign": "center",
        }
      }
    >
      2019
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 2,
        "borderRadius": 3,
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "paddingBottom": 1,
              "paddingTop": 2,
              "textAlign": "center",
            }
          }
        >
          2020
        </Text>
      </View>
    </View>
  </View>
</View>
`;
